enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -1020,24 +1039,68 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
 
 		log.CtxDebugf(ctx, "PublishOperation: stage: %s", stage)
 
-		var response *repb.ExecuteResponse // Only set if stage == COMPLETE
+		// All of these are only set if stage == COMPLETE
+		var response *repb.ExecuteResponse
+		var loggingAux *espb.ExecutionAuxiliaryMetadata
+		var properties *platform.Properties
+		var action *repb.Action
+		var cmd *repb.Command
+		var actionRN *digest.ResourceName
+
 		if stage == repb.ExecutionStage_COMPLETED {
 			response = operation.ExtractExecuteResponse(op)
-		}
-		if response != nil { // The execution completed
-			arn, err := digest.ParseUploadResourceName(taskID)
+
+			// Remove fields that we don't want to write anywhere other than
+			// logs, and save them for later.
```

Comment:
It feels a little cumbersome to have to clear out each of these fields - would it make sense to just clear out all of the auxiliary metadata before storing in the action cache? (I think it would be helpful to keep the isolation_type and scheduling_metadata etc. in the cached execute response, so that we can show these in the UI. e.g. it'd be nice to be able to show why we estimated the task size a certain way)

Reply:
I think that makes sense. Another thing this code does is handle cases where there is more than one ExecutionAuxiliaryMetadata, but maybe I shouldn't worry about that.

Are we concerned about sending any of existing (or future) fields in SchedulingMetadata to bazel via the pubsub send?

Reply:
> Are we concerned about sending any of existing (or future) fields in SchedulingMetadata to bazel via the pubsub send?

I think it's fine to clear out the metadata both before sending to bazel and before storing in the action cache if that's what you mean? (but not before cacheExecuteResponse)

Reply:
Cool, that's what I did.